beekeeper-integration-tests/src/test/java/com/expediagroup/beekeeper/integration/api/BeekeeperApiIntegrationTest.java

```diff
@@ -140,84 +143,124 @@ public void testGetMetadataWhenThereIsFiltering() throws SQLException, Interrupt
-    HousekeepingPath testPath2 = createHousekeepingPath("some_table",
-        "s3://some/path/event_date=2020-01-01/event_hour=0/event_type=B", LifecycleEventType.UNREFERENCED,
-        Duration.parse("P3D").toString());
-    HousekeepingPath testPath3 = createHousekeepingPath("some_table",
-        "s3://some/path/event_date=2020-01-01/event_hour=0/event_type=C", LifecycleEventType.UNREFERENCED,
-        Duration.parse("P3D").toString());
+    HousekeepingPath testPath1 = createHousekeepingPath(someTable, pathA, LifecycleEventType.EXPIRED, duration.toString());
+    HousekeepingPath testPath2 = createHousekeepingPath(someTable, pathB, LifecycleEventType.UNREFERENCED, duration.toString());
+    HousekeepingPath testPath3 = createHousekeepingPath(someTable, pathC, LifecycleEventType.UNREFERENCED, duration.toString());
     testPath1.setHousekeepingStatus(HousekeepingStatus.FAILED);
     testPath1.setCleanupTimestamp(LocalDateTime.parse("1999-05-05T10:41:20"));
     testPath1.setCreationTimestamp(LocalDateTime.parse("1999-05-05T10:41:20"));
-    insertUnreferencedPath(testPath1);
-    insertUnreferencedPath(testPath2);
-    insertUnreferencedPath(testPath3);
 
+    for (HousekeepingPath testPath : Arrays.asList(testPath1, testPath2, testPath3)) {
+      insertUnreferencedPath(testPath);
+    }
     String filters = "?housekeeping_status=FAILED"
         + "&lifecycle_type=EXPIRED"
         + "&deleted_before=2000-05-05T10:41:20"
         + "&registered_before=2000-04-04T10:41:20"
         + "&path=s3://some/path/event_date=2020-01-01/event_hour=0/event_type=A";
 
-    HttpResponse<String> response = testClient.getUnreferencedPaths("some_database", "some_table", filters);
+    HttpResponse<String> response = testClient.getUnreferencedPaths(someDatabase, someTable, filters);
     assertThat(response.statusCode()).isEqualTo(OK.value());
     String body = response.body();
     Page<HousekeepingPathResponse> responsePage = mapper
         .readValue(body, new TypeReference<RestResponsePage<HousekeepingPathResponse>>() {});
     List<HousekeepingPathResponse> result = responsePage.getContent();
-
+    assertThat(responsePage.getTotalElements()).isEqualTo(1L);
     assertThatPathsEqualsResponse(testPath1, result.get(0));
-    assertThat(result.size()).isEqualTo(1);
+    assertThat(result).hasSize(1);
+  }
+
+  @Test
+  public void testPathsPageable() throws SQLException, InterruptedException, IOException {
+
+    HousekeepingPath testPath1 = createHousekeepingPath(someTable, pathA, LifecycleEventType.EXPIRED, duration.toString());
+    HousekeepingPath testPath2 = createHousekeepingPath(someTable, pathB, LifecycleEventType.UNREFERENCED, duration.toString());
+    HousekeepingPath testPath3 = createHousekeepingPath(someTable, pathC, LifecycleEventType.UNREFERENCED, duration.toString());
+
+    for (HousekeepingPath testPath : Arrays.asList(testPath1, testPath2, testPath3)) {
+      testPath.setHousekeepingStatus(HousekeepingStatus.FAILED);
```

Comment:
if the status is relevant for the tests then it can be part of the createHousekeepingPath method and it could have an extra argument